sctp/src/stream/mod.rs

```diff
@@ -269,14 +269,44 @@ impl Stream {
     /// Writes `p` to the DTLS connection with the default Payload Protocol Identifier.
     ///
     /// Returns an error if the write half of this stream is shutdown or `p` is too large.
-    pub fn write(&self, p: &Bytes) -> Result<usize> {
+    pub async fn write(&self, p: &Bytes) -> Result<usize> {
         self.write_sctp(p, self.default_payload_type.load(Ordering::SeqCst).into())
+            .await
+    }
+
+    /// Tries nonblockingly to write `p` to the DTLS connection with the default Payload Protocol Identifier.
+    ///
+    /// Returns an error if the write half of this stream is shutdown or `p` is too large.
+    pub fn try_write(&self, p: &Bytes) -> Result<usize> {
+        self.try_write_sctp(p, self.default_payload_type.load(Ordering::SeqCst).into())
     }
 
     /// Writes `p` to the DTLS connection with the given Payload Protocol Identifier.
     ///
     /// Returns an error if the write half of this stream is shutdown or `p` is too large.
-    pub fn write_sctp(&self, p: &Bytes, ppi: PayloadProtocolIdentifier) -> Result<usize> {
+    pub async fn write_sctp(&self, p: &Bytes, ppi: PayloadProtocolIdentifier) -> Result<usize> {
+        let chunks = self.prepare_write(p, ppi)?;
+        self.send_payload_data(chunks).await?;
+
+        Ok(p.len())
+    }
+
+    /// Tries nonblockingly to write `p` to the DTLS connection with the given Payload Protocol Identifier.
+    ///
+    /// Returns an error if the write half of this stream is shutdown or `p` is too large.
+    pub fn try_write_sctp(&self, p: &Bytes, ppi: PayloadProtocolIdentifier) -> Result<usize> {
```

Comment:
True, but the try_ api turned out to not be necessary. It was born out of a misunderstanding on my part and I left it because I thought it was cool that the sync interface is still possible, albeit in a slightly different form. They can be deleted without affecting the main point of this PR though.

Reply:
thinking about it more. lets provide a single interface

Reply:
I'll remove the try_* API tomorrow then. If a need for it comes up again in the future it's not much work just a bit of plumbing :+1:

Reply:
Done